### ISSUE TITLE
Y001: false positive if listed in `__all__`

### ISSUE THREAD

**Issue by spaceone** [PyCQA/flake8-pyi#347]
`foo.pyi`:
```python
from typing import TypeVar

__all__ = ['T',]

T = TypeVar("T")
```

as `T` is part of `__all__` it shouldn't be detected as `Y001`

**Comment by AlexWaygood** [comment]
Stubs often have complex, version-dependent definitions of `__all__` (e.g. the [stdlib stub for `functools`](https://github.com/python/typeshed/blob/95789d20b267229bf40460df8f3ad02026d849b2/stdlib/functools.pyi#L11)), which makes the question of "what is the value of `__all__` in the stub?" nontrivial to answer.

**Comment by AlexWaygood** [comment]
> Stubs often have complex, version-dependent definitions of `__all__` (e.g. the [stdlib stub for `functools`](https://github.com/python/typeshed/blob/95789d20b267229bf40460df8f3ad02026d849b2/stdlib/functools.pyi#L11)), which makes the question of "what is the value of `__all__` in the stub?" nontrivial to answer.

To flesh this out a little more, there are two possible ways that this question could be answered:

- We could answer it differently depending on which version of Python the plugin is being run with. E.g. we could say "Well, on Python 3.8, `T` isn't included in `__all__`, and you're running flake8 with Python 3.8, so we'll emit Y001 here."
- We could attempt to gather a "composite" definition of `__all__` across all the branches. E.g. if you have an `__all__` definition like this:
  ```python
  import sys
  if sys.version_info >= (3, 8):
      __all__ = ["foo", "baz"]
  else:
      __all__ = ["bar", "baz"]
  ```

  ...then we could say `composite_all = {"foo", "bar", "baz"}`, since `"foo"` is included in `__all__` in the `>=3.8` branch, `"bar"` is included in the `<3.8` branch, and `"baz"` is included in both branches. We could then use the value of `composite_all` to decide whether to emit Y001 when encountering a `TypeVar` that has a name that doesn't start with a leading underscore.

So there are ways of solving this problem. But, I don't much like either of these solutions: they'd both introduce more complexity into our codebase, and they both feel like they could lead to surprising and/or unintuitive behaviour in some situations.

I'm also not convinced that this is a common problem. I've never seen a `TypeVar` name included in an `__all__` definition, and it seems to me like it's pretty rare to import a `TypeVar` from another module. It doesn't seem to me to be too arduous to `# noqa` it if you do have to do it ocassionally.

So -- I'm closing this as "wontfix" :)
